pasqal_cloud/__init__.py

```diff
@@ -44,18 +45,34 @@ def __init__(
         To authenticate to PASQAL Cloud, you have to provide either an
         email/password combination or a TokenProvider instance.
         You may omit the password, you will then be prompted to enter one.
+
+        Also, you can use the parameter `env` to have the endpoints and auth0 endpoints filled automatically.
```

Comment:
This was not done before but since the number of args is starting to pile up, it could be a good opportunity to describe all of them in the docstring and flag the deprecated ones.  

e.g. something like
```python
"""
Function Description

Args:
  username: email of the user to login as
  password: [...]
  [...]
  auth0 (deprecated): Auth0Config object to define the auth0 tenant to target
""" 
```

Maybe put the deprecated arguments last ?

On a side note, I think this sentence here is not very clear. I would say something like "If you want to target a non-prod environment, set the `env` kwarg to the name of the environment to target. This will define automatically for you the auth0 config and the pasqal cloud endpoints.

Reply:
I was thinking about doing it as Batch and Job got big explanations, I'm planning to add it